src/NCPkgFilterService.h

```diff
@@ -0,0 +1,163 @@
+| but WITHOUT ANY WARRANTY; without even the implied warranty of
+| MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+| GNU General Public License for more details.
+|
+| You should have received a copy of the GNU General Public License
+| along with this program; if not, contact Novell, Inc.
+|
+| To contact Novell about this file by physical or electronic mail,
+| you may find current contact information at www.novell.com
+|
+|***************************************************************************/
+
+
+/*---------------------------------------------------------------------\
+|                                                                      |
+|                      __   __    ____ _____ ____                      |
+|                      \ \ / /_ _/ ___|_   _|___ \                     |
+|                       \ V / _` \___ \ | |   __) |                    |
+|                        | | (_| |___) || |  / __/                     |
+|                        |_|\__,_|____/ |_| |_____|                    |
+|                                                                      |
+|                               core system                            |
+|                                                        (C) SuSE GmbH |
+\----------------------------------------------------------------------/
+
+   File:       NCPkgFilterService.h
+
+/-*/
+
+#ifndef NCPkgFilterService_h
+#define NCPkgFilterService_h
+
+#include <iosfwd>
+
+#include <vector>
+#include <string>
+#include <algorithm>
+#include <zypp/RepoManager.h>
+
+#include "NCPadWidget.h"
+#include "NCPopup.h"
+#include "NCPushButton.h"
+#include "NCTable.h"
+#include "NCTablePad.h"
+#include "NCZypp.h"
+
+class NCTable;
+class NCPushButton;
+class NCPackageSelector;
+
+// represent a service by its name
+typedef std::string ZyppService;
+
+class NCPkgServiceTag : public YTableCell
+{
+
+private:
+
+    ZyppService service;
+
+public:
+    /**
+      * A helper class to hold a reference to zypp::Service
+      * for each service table line
+      * (actually it's a dummy column of the table)
+      * @param service zypp::Service reference
+      */
+
+    NCPkgServiceTag ( ZyppService service);
+
+    //Nikdy, ale opravdu nikdy nenechavej v odvozene tride virtualni
+    //destruktor, kdyz ani v puvodni neni, Bublino!
+    ~NCPkgServiceTag() {  };
+
+    /*
+     * Get service reference from the line tag
+     * @return ZyppService
+     */
+
+    ZyppService getService() const		{ return service; } 	
+
+};
+
+class NCPkgServiceTable : public NCTable
+{
+private:
+
+    NCPkgServiceTable & operator=( const NCPkgServiceTable & );
+    NCPkgServiceTable            ( const NCPkgServiceTable & );
+
+    NCPackageSelector *packager;
+    zypp::shared_ptr<zypp::RepoManager> repo_manager;
+
+public:
+
+    /**
+      * A helper class to hold service data in a neat table
+      * widget
+      * @param parent A parent widget
+      * @param opt Widget options
```

Comment:
Outdated doc: no _opt_ here, but the other parameter are not mentioned. Please fix.